apps/prairielearn/src/pages/studentAssessment/studentAssessment.ts

```diff
@@ -164,8 +168,12 @@ router.post(
       });
       res.redirect(req.originalUrl);
     } else if (req.body.__action === 'create_group') {
+      const groupConfig = await getGroupConfig(res.locals.assessment.id);
+      if (!groupConfig.student_authz_create) {
+        throw new HttpStatusError(403, 'You are not authorized to create a group.');
+      }
       await createGroup(
-        req.body.groupName,
+        groupConfig.student_authz_group_name ? req.body.groupName : null,
```

Comment:
Nit: as long as we're here, consider renaming the input to `group_name` to better match our conventions.

Reply:
Done in 3ba9bbb3c0654a91edd3ad9aedae2b834ad0b0b1. Also updated a few IDs around this input name to use kebab-case.